tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -709,38 +753,22 @@ BOOST_AUTO_TEST_CASE(distributed_lgr_II)
-            if (element.partitionType() == Dune::InteriorEntity) {
-                cell_global_ids.push_back(grid.currentData()[1]->globalIdSet().id(element));
-            }
-        }
-        cell_global_ids.shrink_to_fit();
-        auto [all_level_cell_global_ids, displ] = Opm::allGatherv(cell_global_ids, grid.comm());
-        const std::set<int> all_level_cell_global_ids_set(all_level_cell_global_ids.begin(), all_level_cell_global_ids.end());
-        BOOST_CHECK( all_level_cell_global_ids.size() == all_level_cell_global_ids_set.size() );
-
+        refinePatch_and_check(grid, cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
 
         // Check global id is not duplicated for points
-        std::vector<int> point_global_ids;
-        point_global_ids.reserve(grid.currentData()[1]->size(3));
-        for (const auto& point : vertices(grid.levelGridView(1))) {
-            point_global_ids.push_back(grid.currentData()[1]->globalIdSet().id(point));
+        std::vector<int> localPointIds_vec;
+        localPointIds_vec.reserve(grid.currentData().back()->size(3));
+        for (const auto& point : vertices(grid.leafGridView())) {
+            // Notice that all partition type points are pushed back. Selecting only interior points does not bring us to the expected value.
+            localPointIds_vec.push_back(grid.currentData().back()->globalIdSet().id(point));
         }
-        point_global_ids.shrink_to_fit();
-        auto [all_level_point_global_ids, displPoint ] = Opm::allGatherv(point_global_ids, grid.comm());
-        const std::set<int> all_level_point_global_ids_set(all_level_point_global_ids.begin(), all_level_point_global_ids.end());
-        // Notice that cells 8,9, and 10 form a block of dimension 3x1x1.
-        // From the 4x2x2 = 16 parent cell corners, only 4 are partition type interior or border.
-        BOOST_CHECK( all_level_point_global_ids.size()-4 == all_level_point_global_ids_set.size());
-
-        // BOOST_CHECK( all_level_point_global_ids_set.size() == 63 );
-
-        // Commented since communication step is not implemented yet, therefore refined level grids lack of
-        // global ids for non-interior entities, and the leaf cell index of the leaf grid vie
-        // refinePatch_and_check(grid, cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
+        auto [allGlobalIds_points, displPoint ] = Opm::allGatherv(localPointIds_vec, grid.comm());
+        const std::set<int> allGlobalIds_points_set(allGlobalIds_points.begin(), allGlobalIds_points.end());
+
+        // Total global ids in leaf grid view for points: 80 (coarse grid points) + 63 (refined points) - 16 (parent cell corners) = 127
+        // To be included when correction/uniqueness of global ids for points of refined level grid is implemented
+        // std::cout<< allGlobalIds_points_set.size() <<std::endl; // Prints 160 (expected value: 127)
+        //BOOST_CHECK( allGlobalIds_points_set.size() == 127 ); // 127 + 33 = 160
```

Comment:
Same here: Next step?

I will stop commenting here. Please assume the same below whenever you check point ids

Reply:
I will remove the checks that correspond to the general case and keep the ones of fully interior LGRs, if that's okay with you too.